utils/staking-miner/src/dry_run.rs

```diff
@@ -35,18 +36,42 @@ fn force_create_snapshot<T: EPM::Config>(ext: &mut Ext) -> Result<(), Error> {
 }
 
 /// Helper method to print the encoded size of the snapshot.
-fn measure_snapshot_size<T: EPM::Config>(ext: &mut Ext) {
+async fn print_info<T: EPM::Config>(
+	client: &WsClient,
+	ext: &mut Ext,
+	raw_solution: &EPM::RawSolution<EPM::CompactOf<T>>,
+	extrinsic: sp_core::Bytes,
+) where
+	<T as EPM::Config>::Currency: Currency<T::AccountId, Balance = Balance>,
+{
 	ext.execute_with(|| {
-		log::info!(target: LOG_TARGET, "Metadata: {:?}", <EPM::Pallet<T>>::snapshot_metadata());
 		log::info!(
 			target: LOG_TARGET,
-			"Encoded Length: {:?}",
+			"Snapshot Metadata: {:?}",
+			<EPM::Pallet<T>>::snapshot_metadata()
+		);
+		log::info!(
+			target: LOG_TARGET,
+			"Snapshot Encoded Length: {:?}",
 			<EPM::Pallet<T>>::snapshot()
 				.expect("snapshot must exist before calling `measure_snapshot_size`")
 				.encode()
 				.len()
 		);
-	})
+
+		let snapshot_size =
+			<EPM::Pallet<T>>::snapshot_metadata().expect("snapshot must exist by now; qed.");
+		let deposit = EPM::Pallet::<T>::deposit_for(&raw_solution, snapshot_size);
+		log::info!(
+			target: LOG_TARGET,
+			"solution score {:?} / deposit {:?}",
+			&raw_solution.score.iter().map(|x| Token::from(*x)).collect::<Vec<_>>(),
+			Token::from(deposit),
+		);
+	});
+
+	let info = rpc::<Vec<u8>>(client, "payment_queryInfo", params! { extrinsic }).await;
```

Comment:
we're still working on this with @niklasad1, actual return type should be fixed.

Reply:
this is the API https://github.com/paritytech/substrate/blob/master/frame/transaction-payment/rpc/src/lib.rs#L90?

ok, this should return https://docs.rs/pallet-transaction-payment-rpc-runtime-api/3.0.0/pallet_transaction_payment_rpc_runtime_api/struct.RuntimeDispatchInfo.html instead?

basically, if you put `Vec<u8>` here and it returns a `Object/Map` decoding it will fail because of the generics.....

Reply:
Have tried that as well, it also fails. The `Vec<u8>` was an experiment.

Reply:
Okay I've no idea what's happening, now it worked.

Reply:
Otherwise put `JsonValue` here and convert it manually, but based on the [serialize impl in RuntimeDispatchInfo](https://docs.rs/pallet-transaction-payment/3.0.0/src/pallet_transaction_payment/types.rs.html#147) it should be serialized to a `Map/Object`.